torch/csrc/jit/codegen/cuda/index_idgraph_utils.h

```diff
@@ -0,0 +1,155 @@
+#pragma once
+
+#include <torch/csrc/jit/codegen/cuda/fusion.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+// Struct to contain useful information from an index
+//  pass on iterdomain graph.
+struct IndexFromIdGraph {
+  IndexCompute index;
+  std::unordered_map<IterDomain*, Val*> initial_concrete_index_map;
+  std::vector<IterDomain*> resolved_loop_domains;
+
+  explicit IndexFromIdGraph(
+      IndexCompute index,
+      std::unordered_map<IterDomain*, Val*> initial_concrete_index_map,
+      std::vector<IterDomain*> loop_domains);
+};
+
+//! Interface for getting consumer indexing through
+//!  traversal on IterDomain Graph instead of from
+//!  reference replay.
+IndexFromIdGraph getTensorIndexFromIdGraph(
+    const std::vector<kir::ForLoop*>& loops,
+    const TensorView* consumer_tv,
+    const TensorView* producer_tv = nullptr,
+    bool is_global = true,
+    std::unordered_map<IterDomain*, IterDomain*> c2p_map = {});
+
+//! A data structure keeping tack of loop nest dependent indexing
```

Comment:
Can you provide more specific description on what this class is supposed to do and why? It is clear that it keeps track of some info around loop nest s and indexing, but would be really helpful to understand the context more clearly.